tests/nlu/featurizers/test_lm_featurizer.py

```diff
@@ -624,85 +504,60 @@ def test_attention_mask(
             ],
+            [[1, 2], [1], [1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1, 2, 1, 1, 3, 1]],
+        ),
+        (
+            "bert",
+            "bert-base-uncased",
             [
-                ["Good", "even", "ing"],
-                ["hello"],
-                ["you", "re"],
-                ["r", "n", "b"],
-                ["rock", "&", "roll"],
-                [
-                    "here",
-                    "is",
-                    "the",
-                    "sent",
-                    "ence",
-                    "I",
-                    "want",
-                    "embed",
-                    "d",
-                    "ings",
-                    "for",
-                ],
-            ],
-            [
-                [(0, 4), (5, 9), (9, 12)],
-                [(0, 5)],
-                [(0, 3), (4, 6)],
-                [(0, 1), (3, 4), (6, 7)],
-                [(0, 4), (5, 6), (7, 11)],
-                [
-                    (0, 4),
-                    (5, 7),
-                    (8, 11),
-                    (12, 16),
-                    (16, 20),
-                    (21, 22),
-                    (23, 27),
-                    (28, 33),
-                    (33, 34),
-                    (34, 38),
-                    (39, 42),
-                ],
+                "Good evening.",
+                "you're",
+                "r. n. b.",
+                "rock & roll",
+                "here is the sentence I want embeddings for.",
             ],
+            [[1, 1], [1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1, 1, 1, 1, 4, 1]],
         ),
     ],
 )
-@pytest.mark.skip_on_windows
 def test_lm_featurizer_edge_cases(
-    model_name, model_weights, texts, expected_tokens, expected_indices
+    model_name: Text,
+    model_weights: Text,
+    texts: List[Text],
+    expected_number_of_sub_tokens: List[List[float]],
 ):
-
-    if model_weights is None:
-        model_weights_config = {}
-    else:
-        model_weights_config = {"model_weights": model_weights}
-    transformers_config = {**{"model_name": model_name}, **model_weights_config}
-
+    transformers_config = create_config(model_name, model_weights)
     lm_featurizer = LanguageModelFeaturizer(transformers_config)
     whitespace_tokenizer = WhitespaceTokenizer()
 
-    for text, gt_tokens, gt_indices in zip(texts, expected_tokens, expected_indices):
-
+    for i, text in enumerate(texts):
         message = Message.build(text=text)
-        tokens = whitespace_tokenizer.tokenize(message, TEXT)
-        message.set(TOKENS_NAMES[TEXT], tokens)
-        lm_featurizer.process(message)
 
-        assert [t.text for t in tokens] == gt_tokens
-        assert [t.start for t in tokens] == [i[0] for i in gt_indices]
-        assert [t.end for t in tokens] == [i[1] for i in gt_indices]
+        td = TrainingData([message])
+        whitespace_tokenizer.train(td)
+        lm_featurizer.train(td)
```

Comment:
Better to may be test both `train` and `process`?

Reply:
Oh boy you're gonna regret asking for that one ;). I think I found an okay way to do it that doesn't result in too much code duplication but it's a bit of a refactor and the diff looks even more so

Reply:
Haha! Thanks for looking into it. I'll review it end of today or maybe Monday morning latest.

Reply:
Okay, no rush!